src/components/icons/LottieAnimation.tsx

```diff
@@ -1,36 +1,125 @@
 import Player, { LottieRefCurrentProps } from 'lottie-react'
-import React, { useEffect, useRef } from 'react'
+import React, { useEffect, useRef, useState } from 'react'
 
 interface LottieAnimationProps {
   animationData: any
   speed?: number
+  color?: string
   onComplete?: () => void
   style?: React.CSSProperties
 }
 
+/**
+ * Converts hex color to RGBA.
+ *
+ * @param hex - The hex color code.
+ * @returns The RGBA values.
+ */
+const hexToRGBA = (hex: string): number[] => {
+  let r = 0,
+    g = 0,
+    b = 0
+  const a = 1
+  if (hex.length === 7) {
+    r = parseInt(hex.slice(1, 3), 16) / 255
+    g = parseInt(hex.slice(3, 5), 16) / 255
+    b = parseInt(hex.slice(5, 7), 16) / 255
+  }
+  return [r, g, b, a]
+}
+
+/**
+ * Updates colors in a single shape item.
+ *
+ * This function checks if the item represents a stroke or
+ * fill and updates its color using the provided RGBA array.
+ * If the item contains a nested items array, it recursively
+ * calls itself to ensure that all relevant shapes within the
+ * structure are updated with the new color.
+ *
+ * @param item - The shape item object which may contain
+ * stroke or fill color properties.
+ * @param rgbaArray - The RGBA values to set as new color.
+ */
+const updateColorsInItem = (item: any, rgbaArray: number[]) => {
```

Comment:
Let's properly type all Typescript code and avoid the use of `any`. We lose the benefits of the type checker if we use `any`.